compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/window/Dialog.skiko.kt

```diff
@@ -126,12 +128,18 @@ actual fun Dialog(
     properties: DialogProperties,
     content: @Composable () -> Unit
 ) {
+    val blendMode = if (LocalWindowInfo.current.isWindowTransparent) {
+        // Use background alpha channel to respect transparent window shape.
+        BlendMode.SrcAtop
+    } else {
+        BlendMode.SrcOver
```

Comment:
I would add
```
// On a non-transparent window we must use `SrcOver` in order to draw the scrim everywhere,
// even in places where Compose itself doesn't draw anything.
```

Reply:
```suggestion
        // On a non-transparent window default value should be used to draw the scrim everywhere,
        // even in places where Compose itself doesn't draw anything.
        BlendMode.SrcOver
```

Reply:
Moved this condition from dialog to desktop-specific code after discussion with @igordmn